ui/src/app/store/general/slice.ts

```diff
@@ -93,18 +70,79 @@ const generalSlice = createSlice({
     version: generateInitialState(""),
   } as GeneralState,
   reducers: {
-    ...generateGeneralReducers("architectures", "architectures"),
-    ...generateGeneralReducers("bondOptions", "bond_options"),
-    ...generateGeneralReducers("componentsToDisable", "components_to_disable"),
-    ...generateGeneralReducers("defaultMinHweKernel", "default_min_hwe_kernel"),
-    ...generateGeneralReducers("hweKernels", "hwe_kernels"),
-    ...generateGeneralReducers("knownArchitectures", "known_architectures"),
-    ...generateGeneralReducers("machineActions", "machine_actions"),
-    ...generateGeneralReducers("osInfo", "osinfo"),
-    ...generateGeneralReducers("pocketsToDisable", "pockets_to_disable"),
-    ...generateGeneralReducers("powerTypes", "power_types"),
-    ...generateGeneralReducers("version", "version"),
```

Comment:
I'll be honest - I'm completely stumped on how best to handle computed objects/object keys in TypeScript.

In order to get this to work and for TypeScript to be happy I've removed the larger `generateGeneralReducers()` helper util and replaced with four smaller utils, one for each reducer type. This way I can be explicit about exactly what the `reducers` object consists of. It' obviously led to a lot of repetition in the reducer list, but it's at least accurate whereas before it was basically evaluating as `{ [x: string] : ReducerType }`. Anyway I'm open to suggestions to make this better.

Reply:
We've had this same issue with the other slices, so while not ideal this will have to do for now.